docs/SQL/08-schema-design1.md

```diff
@@ -0,0 +1,257 @@
+
+`batches`
+| batch_id | name | start_month |
+|----------|------|-------------|
+
+`instructors`
+| instructor_id | name | email | avg_rating |
+|---------------|------|-------|------------|
+
+`students`
+| student_id | name | email | phone_number | grad_year | univ_name |
+|------------|------|-------|--------------|-----------|-----------|
+
+`classes`
+| class_id | name | schedule_time |
+|----------|------|---------------|
+
+`mentors`
+| mentor_id | name | company_name |
+|-----------|------|--------------|
+
+`mentor_sessions`
+| mentor_session_id | time | duration | student_rating | mentor_rating |
+|-------------------|------|----------|----------------|---------------|
+
+3. **Representing relations:** For understanding this step, we need to look into cardinality.
+
+## Cardinality
+
+When two entities are related to each other, there is a questions: how many of one are related to how many of the other.
+
+For example, for two tables students and batches, cardinality represents how many students are related to how many batches and vice versa.
+
+- 1:1 cardinality means 1 student belongs to only 1 batch and 1 batch has only 1 students.
+- 1:m cardinality means 1 student can belong to multiple batches and 1 batch has only 1 student.
+- m:1 cardinality means 1 student belongs to only 1 batch and 1 batch can have multiple students.
+- m:m cardinality means multiple students can belong to multiple batches, and vice versa.
+
+In cardinality, `1` means an entity can be associated to 1 instance at max, [0, 1]. `m` means an entity can be associated with zero or more instances, [0, 1, 2, ... inf]
+
+### Steps to calculate cardinality
+
+If you want to calculate relationship between `noun1` and `noun2`, then you can do the following:
+ - *Step 1:* If you take one example of `noun2`, how many noun1 are related to this example object. Output : Either `1` or `many`
+ - *Step 2:* If you take one example of `noun1`, how many noun2 are related to this example object. Output : Either `1` or `many`
+ 
+Take output from step1 (o1) and output from step2 (o2). o1:o2 is your relationship.
+
+Let's take an example. 
+What is the cardinality between employee and department. Assume that an employee can be part of only one department. 
+
+ - Step 1: Example of department: Finance. How many employees can be part of Finance. Answer: **many**
+ - Step 2: Example of employee: Sudhanshu. How many department can Sudhanshu be part of? Answer: **one**
+
+So, answer = **many-to-one**
+
+**Example 2:** What is the cardinality between ticket and seat in apps like bookMyShow? 
+
+In one ticket, we can book multiple seats.
+One seat can be booked in only 1 ticket.
+
+So, the final cardinality between ticket and seat is **one-to-many**
+
+**Example 3:**  Consider a monogamous community. What is the cardinality between husband and wife?
+
+| husband | --- married to --- | wife |
+| ------- | ------------------ | ---- |
+| 1       | -->                | 1    |
```

Comment:
Replace the greater than symbol `>` with `&gt;` and the less than symbol `<` with `&lt;`.